packages/api-plugin-fulfillment/src/mutations/updateFulfillmentMethod.js

```diff
@@ -0,0 +1,68 @@
+import SimpleSchema from "simpl-schema";
+import ReactionError from "@reactioncommerce/reaction-error";
+import { FulfillmentMethodSchema } from "../simpleSchemas.js";
+
+const inputSchema = new SimpleSchema({
+  method: FulfillmentMethodSchema,
+  fulfillmentTypeId: String,
+  methodId: String,
+  shopId: String
+});
+
+/**
+ * @method updateFulfillmentMethodMutation
+ * @summary updates Fulfillment method
+ * @param {Object} context - an object containing the per-request state
+ * @param {Object} input - Input object
+ * @param {FulfillmentMethodSchema} input.method - fulfillment method object
+ * @param {String} input.fulfillmentTypeId - id of fulfillment type
+ * @param {String} input.methodId - ff-method Id
+ * @param {String} input.shopId - Shop Id
+ * @returns {Promise<Object>} An object with a `method` property containing the updated method
+ */
+export default async function updateFulfillmentMethodMutation(context, input) {
+  const cleanedInput = inputSchema.clean(input); // add default values and such
+  inputSchema.validate(cleanedInput);
+
+  const { method: inputMethod, methodId, fulfillmentTypeId, shopId } = cleanedInput;
+  const { collections: { Fulfillment } } = context;
+  const method = { ...inputMethod };
+
+  if (!fulfillmentTypeId) throw new ReactionError("invalid-parameter", "Fulfillment Type ID to be updated not provided");
+  if (!methodId) throw new ReactionError("invalid-parameter", "Method ID to be updated not provided");
+
+  await context.validatePermissions(`reaction:legacy:fulfillmentMethods:${methodId}`, "update", { shopId });
+
+  const ffTypeMethodRecord = await Fulfillment.findOne({
+    "_id": fulfillmentTypeId,
+    shopId,
+    "methods._id": methodId
+  });
+  if (!ffTypeMethodRecord) throw new ReactionError("server-error", "Fulfillment Method does not exist");
+
+  // Do not update the fulfillmentType, fulfillmentMethod, group, name & _id fields
+  // Find the matching fulfillmentMethod object and use those values to over-write
+  const currentFulfillmentMethod = (ffTypeMethodRecord.methods || []).find((meth) => meth._id === methodId);
+  if (!currentFulfillmentMethod) throw new ReactionError("server-error", "Fulfillment Method does not exist");
+  const updatedMethod = {
+    ...method,
+    _id: methodId,
+    name: currentFulfillmentMethod.name,
+    group: currentFulfillmentMethod.group,
+    fulfillmentMethod: currentFulfillmentMethod.fulfillmentMethod,
+    fulfillmentType: [ffTypeMethodRecord.fulfillmentType]
```

Comment:
This could be a fragile way to avoid users overwriting unintended props. Instead of using `method: FulfillmentMethodSchema` on the input schema, you could use `method: UserEditableMethodFields`, and then create `UserEditableMethodFields` schema as a subset of `FulfillmentMethodSchema`. That way, after you have called `inputSchema.clean(input)` above, it will be impossible for users to overwrite any fields that they should not. Which means that it would be safe to do:

```js
const updatedMethod = { ...currentFulfillmentMethod, ...method, updatedAt: new Date() };
```

Reply:
Implemented except for updatedAt field.
Awaiting response on the comment [above](https://github.com/reactioncommerce/reaction/pull/6610#discussion_r1018187273)

Reply:
As confirmed by Brent in the above comment, the updatedAt field is only at doc root and not required for each element of the array. So the change is complete. Code will be pushed soon

Reply:
Changes are pushed to branch